src/compile/Component.ts

```diff
@@ -735,6 +747,13 @@ export default class Component {
 					scope = map.get(node);
 				}
 
+				if (node.type === 'LabeledStatement' && node.label.name === '$' && parent.type !== 'Program') {
+					component.warn(node, {
+						code: 'non-top-level-reactive-declaration',
+						message: '$: has no effect outside of the top-level scope'
+					});
+				}
```

Comment:
Out of all the instance script walks, the one in `extract_reactive_store_references` felt the best place to put this. Maybe it should be a separate walk just like in the module script instead?

Reply:
I think it's fine here, at least for now